staging/src/k8s.io/kubectl/pkg/util/completion/completion.go

```diff
@@ -255,6 +259,7 @@ func CompGetFromTemplate(template *string, f cmdutil.Factory, namespace string,
 		return printer.PrintObj, nil
 	}
 
+	f := cmdutil.NewFactory(restClientGetter)
```

Comment:
This is the crucial part to pass through `get` command. OpenAPI, discovery will be reinitialized in here as well.